intellij/src/saros/intellij/ui/views/buttons/AddContactButton.java

```diff
@@ -0,0 +1,167 @@
+import saros.communication.connection.IConnectionStateListener;
+import saros.exceptions.IllegalAWTContextException;
+import saros.exceptions.OperationCanceledException;
+import saros.intellij.ui.Messages;
+import saros.intellij.ui.util.IconManager;
+import saros.intellij.ui.util.SafeDialogUtils;
+import saros.net.ConnectionState;
+import saros.net.xmpp.JID;
+import saros.net.xmpp.contact.XMPPContactsService;
+import saros.repackaged.picocontainer.annotations.Inject;
+
+/** Button to add a contact to the currently connected account. */
+public class AddContactButton extends AbstractToolbarButton {
+  private static final Logger log = Logger.getLogger(AddContactButton.class);
+
+  @Inject private XMPPContactsService xmppContactsService;
+  @Inject private ConnectionHandler connectionHandler;
+
+  private final IConnectionStateListener connectionStateListener =
+      (state, error) -> {
+        if (state == ConnectionState.CONNECTED) {
+          setEnabled(true);
+        } else {
+          setEnabled(false);
+        }
+      };
+
+  public AddContactButton(@NotNull Project project) {
+    super(project, null, Messages.AddContactButton_tooltip, IconManager.ADD_CONTACT_ICON);
+
+    SarosPluginContext.initComponent(this);
+
+    connectionHandler.addConnectionStateListener(connectionStateListener);
+
+    addActionListener(actionEvent -> showAddContactDialog());
+
+    if (!connectionHandler.isConnected()) {
+      setEnabled(false);
+    }
+  }
+
+  @Override
+  public void dispose() {
+    connectionHandler.removeConnectionStateListener(connectionStateListener);
+  }
+
+  /**
+   * Shows the dialogs requesting the JID and (optional) nickname for the account to add as a
+   * contact.
+   *
+   * <p>Calls {@link XMPPContactsService#addContact(JID, String, BiPredicate)} if the dialogs were
+   * not canceled.
+   */
+  private void showAddContactDialog() {
+
+    // TODO better communicate issue to user
+    // TODO use checkInput as it disables the OK button instead of just doing nothing when OK is
+    //  clicked on illegal input.
+    InputValidator inputValidator =
+        new InputValidator() {
+          @Override
+          public boolean checkInput(String inputString) {
+            return true;
+          }
+
+          @Override
+          public boolean canClose(String inputString) {
+            JID jid = new JID(inputString);
+
+            return jid.isValid() && !jid.getName().isEmpty();
+          }
+        };
+
+    String userID;
+
+    try {
+      userID =
+          SafeDialogUtils.showInputDialog(
+              project,
+              Messages.AddContactButton_contact_jid_dialog_message,
+              Messages.AddContactButton_contact_jid_dialog_initial_input,
+              Messages.AddContactButton_contact_jid_dialog_title,
+              inputValidator,
+              new TextRange(0, 0));
+
+    } catch (IllegalAWTContextException e) {
+      log.error("Account creation failed.", e);
```

Comment:
The account creation has not even started here. Can this exception even happen on normal user interaction. Reading the Javadoc for the exception this could only happen if we somehow manage to call this code inside a write access which currently seems impossible for me.

TLDR: I guess the TODO can be removed.

Reply:
Well, the user pressed the button and nothing happened. So it would be nice to have some kind of feedback (especially as it will enable the user to make a more accurate bug report than just "nothing happened when I pressed the button"). But yes, the log message is misleading.

Regarding whether this can actually happen: I don't think so, but I don't think it hurts to cover all bases in this case.